relay-event-schema/src/protocol/user.rs

```diff
@@ -202,6 +202,19 @@ mod tests {
         assert_eq!(output, user.to_json().unwrap());
     }
 
+    #[test]
+    fn test_user_lenient_username() {
+        let input = r#"{"username":42}"#;
+        let output = r#"{"username":"42"}"#;
+        let user = Annotated::new(User {
+            username: Annotated::new("42".to_string().into()),
+            ..User::default()
+        });
+
+        assert_eq!(user, Annotated::from_json(input).unwrap());
+        assert_eq!(output, user.to_json().unwrap());
```

Comment:
imo the round trip is enough

```rs
assert_eq!( Annotated::from_json(input).unwrap().to_json().unwrap(), r#"{"username":"42"}"#)
```

Can also maybe just use insta here to assert the output?

Reply:
I copy-pasted the existing tests around this, would like to keep them as similar as possible.